### PR TITLE
fix frontend tests; minor readme update

### PR DESCRIPTION
**Pull request by tobitege**
Attempt to fix several test files for the frontend folder, one file remains an issue.

The first run left 23 tests failed and 1 error:
grafik

This PR fixes the error and gets down to 6 failed tests, mainly for the SettingsModal file.
grafik

Someone with more expertise in Vite/React can hopefully find the cause for issues like this:
grafik

I hope the experts take a look at these. :)
@rbren @amanape

### CODE REVIEW

File: frontend/src/components/chat/ChatInput.test.tsx [hunk 1]
```diff
@@ -47,26 +47,29 @@ describe("ChatInput", () => {
...
 
   it("should be able to send a message when the enter key is pressed", () => {
     const { getByRole } = render(<ChatInput onSendMessage={onSendMessage} />);
     const textarea = getByRole("textbox");
 
-    act(() => {
-      userEvent.type(textarea, "Hello, world!{enter}");
-    });
+	fireEvent.change(textarea, { target: { value: "Hello, world!" } });
+    fireEvent.keyDown(textarea, { key: "Enter", code: "Enter", charCode: 13 });
```
Comment (rbren) [inline frontend/src/components/chat/ChatInput.test.tsx:72]: indent is off here

Comment (tobitege) [reply frontend/src/components/chat/ChatInput.test.tsx]: Thanks, slipped by me. Fixed now.

**Comment by tobitege** [comment]
Oh, missed to run local linter, will fix asap.

**Comment by tobitege** [comment]
Added a couple more lint fixes that came up via make lint (not my changes 😁 ).

**Comment by iFurySt** [comment]
LGTM

**Comment by tobitege** [comment]
Hello @li-boxuan,
would you have time to do a review? 🤗

**Comment by li-boxuan** [comment]
@tobitege I'd love to but I can't; I am a frontend newbie :)

**Comment by tobitege** [comment]
> @tobitege I'd love to but I can't; I am a frontend newbie :)

Same! Fair enough :)

File: frontend/src/components/file-explorer/FileExplorer.test.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import { waitFor, screen } from "@testing-library/react";
+import { waitFor } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
 import { act } from "react-dom/test-utils";
```
Comment (amanape) [inline frontend/src/components/file-explorer/FileExplorer.test.tsx:4]:
```suggestion
import { waitFor, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
```

This should fix: `Warning: ReactDOMTestUtils.act is deprecated in favor of React.act. Import act from react instead of react-dom/test-utils. See https://react.dev/warnings/react-dom-test-utils for more info.`

File: frontend/src/components/chat/ChatInterface.test.tsx [hunk 3]
```diff
@@ -76,9 +75,9 @@ describe("ChatInterface", () => {
     });
 
     const input = screen.getByRole("textbox");
-    act(() => {
-      userEvent.type(input, "my message{enter}");
-    });
+    // await userEvent.type(input, "my message{enter}");
```
Comment (amanape) [inline frontend/src/components/chat/ChatInterface.test.tsx:78]: Remove comment

**Review by amanape — CHANGES_REQUESTED** [review]
Looks good! Thanks for fixing the test, just a little miss with the `act` as you've done for the `ChatInterface` and `BaseModal` tests.

Also, you have a lot of mixed usage with user event handling. Some are `await userEvent ...`, others remain wrapped in `act` but with `async...await`, some replaced with the more basic `fireEvent`, and some tests include all of them.

Generally it would be preferable to use one way for consistency unless there is another reason to do so. `userEvent` is generally better over `fireEvent` because it simulates user events more realistically ([Differences from `fireEvent`](https://testing-library.com/docs/user-event/intro/#differences-from-fireevent)).

If you would be up for it, whether in this PR or another, you can update the `@testing-library/user-event` package to the latest version and the API will be more straightforward than the current version (`const user = userEvent.setup()` -> `await user.click(...)`

**Comment by amanape** [comment]
As for the tests related to the settings modal, they are somewhat frustrating to test and update due to the nextui select inputs. Sometimes you must test for the key rather than the value, and sometimes the test DOM does not recognize the dropdown that is opened. I can try to look into later but maybe this can give you some hints (if it is related anyways)

**Comment by tobitege** [comment]
> As for the tests related to the settings modal, they are somewhat frustrating to test and update due to the nextui select inputs. Sometimes you must test for the key rather than the value, and sometimes the test DOM does not recognize the dropdown that is opened. I can try to look into later but maybe this can give you some hints (if it is related anyways)

Thanks so much for all your feedback, appreciated!
The tests are inconveniently difficult to get in line, especially when .type is involved, which is asynchronous and can cause "my message" to come back as "mmyy  mmeessaaggee" or completely garbled up. That's the point where I'd use FireEvent instead for the value.

TL;DR: set back to Draft, will give it another go to clean it up more (bring in line across files).

**Comment by amanape** [comment]
> > As for the tests related to the settings modal, they are somewhat frustrating to test and update due to the nextui select inputs. Sometimes you must test for the key rather than the value, and sometimes the test DOM does not recognize the dropdown that is opened. I can try to look into later but maybe this can give you some hints (if it is related anyways)
> 
> Thanks so much for all your feedback, appreciated! The tests are inconveniently difficult to get in line, especially when .type is involved, which is asynchronous and can cause "my message" to come back as "mmyy mmeessaaggee" or completely garbled up. That's the point where I'd use FireEvent instead for the value.
> 
> TL;DR: set back to Draft, will give it another go to clean it up more (bring in line across files).

Yes, these are the scenarios you'd prefer to use `fireEvent` over `userEvent` 👍 

Though we should update the the latest versions of the testing libraries (user event especially), and see if the issue persists.

**Comment by tobitege** [comment]
All tests passed! 😀  Trick in SettingsModal was using await/async.

grafik

**Comment by amanape** [comment]
Awesome! Thanks fixing them up!

Now we just have to keep it that way 😅

**Review by amanape — APPROVED** [review]
